internal/worker/jobworker/jobworker.go

```diff
@@ -0,0 +1,186 @@
+package jobworker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"sync"
+	"syscall"
+
+	"github.com/joshuarubin/teleport-job-worker/internal/config"
+	"github.com/joshuarubin/teleport-job-worker/internal/job"
+	"github.com/joshuarubin/teleport-job-worker/internal/user"
+	"github.com/joshuarubin/teleport-job-worker/internal/worker"
+)
+
+// ReexecCommand contains the necessary configuration for the JobWorker to be
+// able to call exec.Command() and expect that the current binary will be
+// reexecuted, the JobWorker will be reinstantiated, and
+// JobWorker.StartJobChild() will be called with the remaining arguments passed
+// in as command and args
+type ReexecCommand struct {
+	Command string   // often "/proc/self/exe"
+	Args    []string // usually a command that puts this binary in a different "mode", e.g. "child" or "helper"
+	Env     []string // additional env variables, in the form of "key=value", to be added to the current os.Environ()
```

Comment:
Ditch Env?

Reply:
It was required for testing unfortunately